api/src/main/java/io/grpc/CallOptions.java

```diff
@@ -403,8 +471,9 @@ public CallOptions withMaxInboundMessageSize(int maxSize) {
   @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2563")
   public CallOptions withMaxOutboundMessageSize(int maxSize) {
     checkArgument(maxSize >= 0, "invalid maxsize %s", maxSize);
-    CallOptions newOptions = new CallOptions(this);
-    newOptions.maxOutboundMessageSize = maxSize;
+    CallOptions newOptions = fullBuild(this)
+            .maxOutboundMessageSize(maxSize)
+            .build();
     return newOptions;
```

Comment:
The local var `newOptions` seems unnecessary. You could just do:

```
return fullBuild(this)
            .maxOutboundMessageSize(maxSize)
            .build();
```

Same for all other similar usage elsewhere